src/util/ecdsa.rs

```diff
@@ -413,6 +415,72 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+/// An ECDSA signature with the corresponding hash type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EcdsaSig {
+    /// The underlying ECDSA Signature
+    pub sig: secp256k1::Signature,
+    /// The corresponding hash type
+    pub hash_ty: EcdsaSigHashType,
+}
+
+impl EcdsaSig {
+
+    /// Deserialize from slice
+    pub fn from_slice(sl: &[u8]) -> Result<Self, EcdsaSigError> {
+        let (hash_ty, sig) = sl.split_last()
+            .ok_or(EcdsaSigError::EmptySignature)?;
+        let hash_ty = EcdsaSigHashType::from_u32_standard(*hash_ty as u32)
+            .map_err(|_| EcdsaSigError::NonStandardSigHashType(*hash_ty))?;
```

Comment:
May be nicer to match on `NonStandardSigHashType` (`map_err(|NonStandardSigHashType| EcdsaSigError::NonStandardSigHashType(*hash_ty)`) to protect against possible future refactors but not a big deal.